### PR TITLE
feature(logbook-spring-boot-autoconfigure): add support for logbook-openfeign

### PR DESCRIPTION
**Pull request by p-8-z**
Add support for `logbook-openfeign` in `logbook-spring-boot-starter` package.

## Detailed Description
Add Conditional `FeignLogbookLoggerConfiguration` to `LogbookAutoConfiguration` to enable auto-configuration of `FeignLogbookLogger` class.

### CODE REVIEW

**Review by msdousti — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution.
In addition to addressing comments above, could you please rebase on the latest main branch? (Needed for the pipeline to succeed).

File: logbook-spring-boot-autoconfigure/pom.xml [hunk 1]
```diff
@@ -64,6 +64,17 @@
...
+        <dependency>
+            <groupId>org.zalando</groupId>
+            <artifactId>logbook-openfeign</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>io.github.openfeign</groupId>
+            <artifactId>feign-core</artifactId>
+            <version>${feign.version}</version>
+            <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (p-8-z) [reply logbook-spring-boot-autoconfigure/pom.xml:76]: I was not sure with this one, but I think your right.

File: logbook-spring-boot-autoconfigure/pom.xml [hunk 2]
```diff
@@ -64,6 +64,17 @@
...
+        <dependency>
+            <groupId>org.zalando</groupId>
+            <artifactId>logbook-openfeign</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>io.github.openfeign</groupId>
+            <artifactId>feign-core</artifactId>
+            <version>${feign.version}</version>
+            <scope>optional</scope>
```
Comment (msdousti) [inline logbook-spring-boot-autoconfigure/pom.xml:76]: Thanks for updating this. Maven has 6 scopes, but `optional` is [not one of them](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope). Optional is actually a flag, which can be true or false.
```suggestion
            <optional>true</optional>
```

**Review by msdousti — CHANGES_REQUESTED** [review]
Please apply the comment and also make sure your branch is up to date with main.

**Comment by msdousti** [comment]
👍

**Pull request merged**